File: aspnet-in-practice/08-09/CS/CoolMVBlog/Utils/Filters/LoadTagCloudAttribute.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;
using CoolMvcBlog.Models;

namespace CoolMvcBlog.Utils.Filters
{
    public class LoadTagCloudAttribute : ActionFilterAttribute
    {
        public override void OnResultExecuting(ResultExecutingContext filterContext)
        {
            base.OnResultExecuting(filterContext);
            
            var view = filterContext.Result as ViewResult;
            if (view == null)
                return;

            var model = view.ViewData.Model as IHasTagCloud;
            if (model == null)
                return;

            using (var ctx = new BlogModelContainer())
            {
                var service = new TagCloudService(ctx);

                model.TagCloudItems = service.GetTagCloudItems();
            }
        }
    }
}

File: aspnet-in-practice/08-09/VB/CoolMVCBlog/Utils/Filters/LoadTagCloudAttribute.vb
Public Class LoadTagCloudAttribute
    Inherits ActionFilterAttribute

    Public Overrides Sub OnResultExecuting(ByVal filterContext As ResultExecutingContext)
        MyBase.OnResultExecuting(filterContext)

        Dim view = TryCast(filterContext.Result, ViewResult)
        If view Is Nothing Then
            Return
        End If

        Dim model = TryCast(view.ViewData.Model, IHasTagCloud)
        If model Is Nothing Then
            Return
        End If

        Using ctx As New BlogModelContainer
            Dim service = New TagCloudService(ctx)
            model.TagCloudItems = service.GetTagCloudItems
        End Using

    End Sub

End Class